src/test/java/com/google/devtools/build/lib/blackbox/tests/manageddirs/ManagedDirectoriesBlackBoxTest.java

```diff
@@ -392,6 +377,39 @@ public void testWorkspaceSymlinkThrowsWithManagedDirectories() throws Exception
         .isTrue();
   }
 
+  @Test
+  public void testNoCheckFiles(@TestParameter boolean allSkips, @TestParameter boolean watchFs) throws Exception {
+    generateProject();
+    buildExpectRepositoryRuleCalled(false, watchFs);
+    checkProjectFiles();
+
+    Path nodeModules = context().getWorkDir().resolve("node_modules");
+    assertThat(nodeModules.toFile().isDirectory()).isTrue();
+    PathUtils.deleteTree(nodeModules);
+    assertThat(nodeModules.toFile().isDirectory()).isFalse();
+
+    // We don't check that the external file disappeared so the build is broken
```

Comment:
I don't understand this statement. Why is the build broken, as opposed to spuriously succeeding because we don't actually execute any actions that depend on the missing files? What causes us to try to rebuild anything at all? I would expect that the missing files would go unnoticed until an in-repo file was modified that caused a downstream action to re-run, with that downstream action consuming a file from the deleted managed directory.

That being said, I'm also fine with your just deleting this test. Managed directories is a super gnarly feature, and I'm satisfied that your flag isn't making anything worse.

Reply:
I was writing this comment comparing to the above test `testNodeModulesDeleted` which notices that we deleted this file and the rebuild fixes it. I updated the comment to reference it. Do you think it is still useful to keep this around like this?